src/EventStore.Client/StreamRevision.cs

```diff
@@ -10,6 +10,11 @@ namespace EventStore.Client {
 		public static StreamRevision FromInt64(long value) =>
 			value == -1 ? None : new StreamRevision(Convert.ToUInt64(value));
 
+		public static StreamRevision FromStreamPosition(StreamPosition position) => position.ToUInt64() switch {
+			ulong.MaxValue => throw new ArgumentOutOfRangeException(nameof(position)),
```

Comment:
Minor: You could switch on `position` and match `StreamPosition.End`? Less magic values.

Reply:
You can only switch on compile time constants with switch expressions. The alternative is to move this to `StreamPosition.ToStreamRevision` where we do have access to the constants.